File: src-1.x/BYTES.NET/MEF/API/Metadata.vb
'import .net namespace(s) required
Imports System.ComponentModel.Composition

Namespace MEF.API

    ''' <summary>
    ''' the basic (extension) metadata class 
    ''' </summary>
    ''' <remarks>based on the article found at 'https://stefanhenneken.wordpress.com/2011/06/05/mef-teil-2-metadaten-und-erstellungsrichtlinien'</remarks>
    <MetadataAttribute>
    <AttributeUsage(AttributeTargets.Class)>
    Public Class Metadata

        Inherits ExportAttribute

#Region "public properties"

        Public Property ID As String

        Public ReadOnly Property InterfaceType As Type
            Get

                Return MyBase.ContractType

            End Get
        End Property

#End Region

#Region "public new instance method(s)"

        ''' <summary>
        ''' default new instance method
        ''' </summary>
        ''' <param name="interfaceType"></param>
        Public Sub New(ByVal interfaceType As Type)

            MyBase.New(interfaceType)

        End Sub

#End Region

    End Class

End Namespace

File: src-2.x/BYTES.NET/IO/Metadata.cs
//import .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BYTES.NET.IO
{
    [AttributeUsage(AttributeTargets.Class)]
    public class Metadata : Attribute
    {
        public string Name { get; set; }

        public string[] Aliases { get; set; }

        public Metadata()
        {
            this.Aliases = new string[] { };
        }
    }
}
